controllers/tykoasapidefinition_controller.go

```diff
@@ -411,6 +622,41 @@ func deleteOasApi(ctx context.Context, tykOASCrd *v1alpha1.TykOasApiDefinition)
 	return nil
 }
 
+func (r *TykOasApiDefinitionReconciler) updateBaseOasAPIVersionsStatus(
+	ctx context.Context,
+	log logr.Logger,
+	tykOASCrd *v1alpha1.TykOasApiDefinition,
+) error {
+	if tykOASCrd == nil {
+		return nil
+	}
+
+	if tykOASCrd.Spec.Versioning.Enabled {
+		for _, v := range tykOASCrd.Spec.Versioning.Versions {
+			var versionOasAPI v1alpha1.TykOasApiDefinition
+
+			if err := r.Get(ctx, types.NamespacedName{
+				Name:      v.OasApiDefinitionRef,
+				Namespace: v.Namespace,
+			}, &versionOasAPI); err != nil {
+				if k8serrors.IsNotFound(err) {
+					log.Info("failed to get versioned TykOasApiDefinition CR", "err", err.Error())
+				}
+
+				return err
+			}
```

Comment:
Please correct me if i am wrong. This logic only runs while reconciling the delete request and I couldn't get why non-existent OAS API Definitions on k8s prevents this resource from deletion?